libnetwork/sandbox_linux.go

```diff
@@ -157,14 +157,35 @@ func (sb *Sandbox) SetKey(basePath string) error {
 		}
 	}
 
+	sb.ipv6Disabled = sb.calcIPv6Disabled(osSbox)
+
+	// Finish configuration, based on the newly-discovered availability of IPv6
+	// in the container.
+	if err := sb.ctrPrestartDNS(); err != nil {
+		return err
+	}
+
 	for _, ep := range sb.Endpoints() {
 		if err = sb.populateNetworkResources(ep); err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
 
+// Set Sandbox.ipv6Disabled to its sysctl'd value (which runC hasn't yet
+// applied), else according to whether the container's 'lo' interface has
+// an IPv6 address ('::1' is assigned by the kernel when IPv6 is enabled).
+func (sb *Sandbox) calcIPv6Disabled(osSbox *osl.Namespace) bool {
+	for k, v := range sb.config.sysctls {
+		if k == "net.ipv6.conf.all.disable_ipv6" {
+			return v == "1"
+		}
+	}
```

Comment:
Ok, we talked each other around on doing the work in SetKey to make sure buildkit is ok! (It makes things a lot simpler for follow-up changes to know that the task-created code will run). And, this was the only way try to read the user's intention at that point ... but, I agree it's got holes that only documentation would dig us out of. Not great.

So, I'll try to find a way for the daemon to promise to libnetwork that it'll make a call between task create/start where it can finalise config, without  Buildkit making that promise.

Then, for buildkit, libnetwork can finish its config during SetKey instead. As you say, luckily there's no way to set sysctls in buildkit's case.